src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemFactory.java

```diff
@@ -114,12 +141,13 @@ static CgroupMetrics create() {
             anyControllersEnabled = anyControllersEnabled || info.isEnabled();
         }
 
-        // If there are no mounted controllers in mountinfo, but we've only
-        // seen 0 hierarchy IDs in /proc/cgroups, we are on a cgroups v1 system.
+        // If there are no mounted, relevant cgroup controllers in mountinfo and only
+        // 0 hierarchy IDs in /proc/cgroups have been seen, we are on a cgroups v1 system.
         // However, continuing in that case does not make sense as we'd need
-        // information from mountinfo for the mounted controller paths anyway.
+        // information from mountinfo for the mounted controller paths which we wouldn't
+        // find anyway in that case.
         try (Stream<String> mntInfo = CgroupUtil.readFilePrivileged(Paths.get(mountInfo))) {
-            boolean anyCgroupMounted = mntInfo.anyMatch(line -> line.contains("cgroup"));
+            boolean anyCgroupMounted = mntInfo.anyMatch(CgroupSubsystemFactory::noSystemdCgroupLine);
```

Comment:
I'd prefer a similar approach to the hotspot side where you do a positive check for controllers we are interested in rather than a negative check for systemd.

Reply:
OK. I'll probably fold JDK-8254001 into this one then.

Reply:
@bobvandette Done in the latest version. Thoughts?

Reply:
I'm a little uncomfortable making such a dramatic change just to fix this small isolated problem.   I think about all the churn we've had on these files and the backports to come.  Could you just add a CgroupSubsystemFactory::isValidSubsystem function which checks for cpuset, cpuacct, memory, etc instead of the major change?

Reply:
Thanks for having a look! Sure, we could do that instead. I'm not sure this is the best solution going forward, though. Thinking about JDK 17...

Anyway, I'll look into a more conservative fix.